src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.ValidateSignature.cs

```diff
@@ -170,36 +173,36 @@ private static (SignatureValidationResult? validResult, bool KidMatched, KeyMatc
             bool kidExists = !string.IsNullOrEmpty(jwtToken.Kid);
             bool kidMatched = false;
             IList<SecurityKey>? keysAttempted = null;
-            IList<SignatureValidationResult>? results = null;
+            IList<ExceptionDetail>? errors = null;
 
             for (int i = 0; i < keysList.Count; i++)
             {
                 SecurityKey key = keysList[i];
-                SignatureValidationResult result = ValidateSignatureWithKey(jwtToken, key, validationParameters, callContext);
-                if (result.IsValid)
+                Result<SecurityKey, ExceptionDetail> result = ValidateSignatureWithKey(jwtToken, key, validationParameters, callContext);
+                if (result.IsSuccess)
```

Comment:
null check on `result`?

Reply:
Value type explained in another comment.